provider/openstack/firewaller.go

```diff
@@ -108,22 +435,14 @@ func (c *defaultFirewaller) SetUpGroups(controllerUUID, machineId string, apiPor
 	if err != nil {
 		return nil, err
 	}
-	groups := []neutron.SecurityGroupV2{jujuGroup, machineGroup}
+	groups := []nova.SecurityGroupName{{jujuGroup.Name}, {machineGroup.Name}}
 	if c.environ.ecfg().useDefaultSecurityGroup() {
-		// Security Group Names in Neutron do not have to be unique.  This
-		// function returns an array
-		defaultGroups, err := c.environ.neutron().SecurityGroupByNameV2("default")
-		if err != nil {
-			return nil, fmt.Errorf("loading default security group: %v", err)
-		}
-		for _, defaultGroup := range defaultGroups {
-			groups = append(groups, defaultGroup)
-		}
+		groups = append(groups, nova.SecurityGroupName{"default"})
```

Comment:
I see no reference in the nova api docs as to how creating a server actually handles being given a group name that related to multiple groups, so I guess I'm fine with either behaviour here.